src/main/java/com/unityTest/courseManagement/utils/SourceTypeConverter.java

```diff
@@ -0,0 +1,17 @@
+package com.unityTest.courseManagement.utils;
+
+import com.unityTest.courseManagement.entity.SourceType;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.stereotype.Component;
+
+/**
+ * Converts source types to upper case when used in api calls. Allows use of lower case conventions
+ * in path naming.
+ */
+@Component
+public class SourceTypeConverter implements Converter<String, SourceType> {
```

Comment:
Could we not put this function in the `Utils.java` class? Seems an overkill having a class dedicated to one function. I don't know enough about these frameworks to give any definitive suggestion.

Reply:
The key here is the `@Component` annotation. It lets me hook the functionality into **any** endpoint that uses `SourceType` in it so that it can be lower case in this situation. The function itself is not really that big of a deal if that makes sense.

Reply:
@OZoneGuy Thanks!